app/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorFragment.kt

```diff
@@ -136,26 +137,28 @@ class SurveySelectorFragment : AbstractFragment(), BackPressListener {
 
   @Suppress("UnrememberedMutableState")
   private fun showRemoveConfirmationDialogs(onConfirm: () -> Unit) {
-    binding.composeView.apply {
-      setContent {
-        var showRemoveWarningDialog by mutableStateOf(true)
+    val dialogComposeView =
+      ComposeView(requireContext()).apply {
+        setContent {
+          var showRemoveWarningDialog by mutableStateOf(true)
 
-        AppTheme {
-          if (showRemoveWarningDialog) {
-            ConfirmationDialog(
-              title = R.string.remove_offline_access_warning_title,
-              description = R.string.remove_offline_access_warning_dialog_body,
-              confirmButtonText = R.string.remove_offline_access_warning_confirm_button,
-              signOutCallback = {
-                onConfirm()
-                showRemoveWarningDialog = false
-              },
-              dismissCallback = { showRemoveWarningDialog = false },
-            )
+          AppTheme {
+            if (showRemoveWarningDialog) {
+              ConfirmationDialog(
+                title = R.string.remove_offline_access_warning_title,
+                description = R.string.remove_offline_access_warning_dialog_body,
+                confirmButtonText = R.string.remove_offline_access_warning_confirm_button,
+                signOutCallback = {
+                  onConfirm()
+                  showRemoveWarningDialog = false
+                },
+                dismissCallback = { showRemoveWarningDialog = false },
+              )
+            }
           }
         }
       }
-    }
+    (view as ViewGroup).addView(dialogComposeView)
```

Comment:
Why do we need to add dialog to root view?

Reply:
else it will not be visible on top, right?

Reply:
It would. Can you try once without this?

Reply:
Ok, my bad